### DISCUSSION TITLE
Geonode installation alternatives

### DISCUSSION THREAD

**Discussion by azizOthmani** [GeoNode/geonode#12008] [Q&A]
Hello everyone, I want to know what the differences are between the basic installation of GeoNode and the advanced one. Additionally, what are the differences between GeoNode Vanilla and GeoNode Project? Which one should I install, and is it recommended to install GeoNode with Docker?

**Comment by gromit83** [comment]
Thinking about the same. The documentation is full of holes and if you follow it religiously it looks like a lot of the backend like django have changed ALOT since they where made and a lot of the classes that django is callig for is made obsolete and changed with other calls. 

Good example is: ImportError: cannot import name 'ugettext_lazy' from 'django.utils.translation' (/home/my_geonode/.virtualenvs/geonode/lib/python3.10/site-packages/django/utils/translation/__init__.py)

Think the maintaner should try and install geonode from the master guide themselves on a fresh ubuntu and see it doesn't work and requires A LOT of bugsquashing.

**Comment by ridoo** [comment]
Hey, I am not so new anymore, but far from the experiences of the other contributors here. I can feel you regarding the installation process.

However, we already had a lot of discussion around this topic[^1][^2][^3]. You always have to bear in mind that GeoNode is more than ten years old! Of course there is some legacy. But by contributing you have the chance to make things better.

My current take is to encapsulate lots of the complexity in a customized GeoNode docker setup. This, for now, needs some manual foo to track upstream, but to me it felt a lot easier to handle. [Read here](https://github.com/GeoNodeUserGroup-DE/geonode-blueprint-docker?tab=readme-ov-file#background) for the whole background.

We (the [German GeoNode UserGroup](https://github.com/orgs/GeoNodeUserGroup-DE)) tested the [GeoNode blueprint](https://github.com/GeoNodeUserGroup-DE/geonode-blueprint-docker) in [the Thuenen Atlas](https://github.com/Thuenen-GeoNode-Development/thuenen_atlas) already, and I use this blueprint for more and more projects. I am looking forward to bring this to upstream (I [saw your comment](https://github.com/GeoNode/geonode-project/issues/471#issuecomment-2015527117) @giohappy) , but I am a bit hesitant to add yet another way to install GeoNode. 

Maybe, there are untested use cases. So handle with care. Also, the project setup still is not a no-brainer! You have to know things how to stick together components and code within docker compose. However, in the end, IMHO it is written more from the docker perspective and provides you with a devcontainer setup for your project.

Feel free to test things out and [give feedback](https://github.com/GeoNodeUserGroup-DE/geonode-blueprint-docker/issues).

[^1]: https://github.com/GeoNode/geonode-project/issues/471
[^2]: https://github.com/GeoNode/geonode-project/discussions/460
[^3]: https://lists.osgeo.org/pipermail/geonode-devel/2023-August/003335.html

**Comment by giohappy** [comment]
I remind everyone that GeoNode is a community project. We are all aware that its documentation contains legacy and outdated parts. Any contribution to fixing them and improving the documentation is always welcome!

We normally use the Docker approach (and recently k8s) to install GeoNode.
With Docker, a basic setup is a matter of running:

```shell
python create-envfile.py
docker compose up -d
```

This works both for the vanilla GeoNode and a project generated from the geonode-project template.

@ridoo keep us updated about the blueprint. We already discussed the approach in the past, and for our use cases it was lacking some features, but we're open to adopt different strategies if they can improve the process.

**Comment by t-book** [comment]
@azizOthmani Regarding your answer here is an explanation of terms:

https://docs.geonode.org/en/master/install/basic/index.html

**Basic Installation** = Using docker to run preconfigured services
**Advances Installation** = Installing GeoNode without docker but f.e. from apt packages. This is much harder as you need to know how everything is wired together

**Vanilla Installation** = is the acual geonode code that you see in this repository. This could also be namend "core"
**GeoNode Project** = Is a wrapper around this repository. A dedicated Django project that is there to start with own development. (Without touching the core code)

For completness there also was **SPC** geonode (discontinued), which also was a docker setup approach similar to the official. 

From what I understand so far @ridoo created an alternative docker project like "geonode Project". Is that right @ridoo? And can we read somewhere about the main goals, milestones, approvements from your setup? (sorry if I missed a discussion)

**Reply by gromit83** [reply]
I basically followed the vanilla guide and the code that it downloads uses outdated django calls. So it's a mess to try to wrangle. So i'm trying the german docker approach. I'm not a fan of the all inclusive docker as I would like to have geoserver and postgis on different VMS for redundancy and easy testing and thus keeping other data safe even if I bork the geonode.

I'm alas no coder, but rather a mediocre linux user :)

**Reply by t-book** [reply]
"downloads uses outdated django calls." is what I do not understand. As you pull from master? So it should be latest code.

**Reply by t-book** [reply]
"ImportError: cannot import name 'ugettext_lazy' from 'django.utils.translation'" sounds you have django 2 code but 3 installed

**Reply by ridoo** [reply]
> From what I understand so far @ridoo created an alternative docker project like "geonode Project". Is that right @ridoo? And can we read somewhere about the main goals, milestones, approvements from your setup? (sorry if I missed a discussion)

@t-book The motivation behind the blueprint is the essence from the [(long) discussion](https://github.com/GeoNode/geonode-project/issues/471) around the deduplication of GeoNode images and [my root question](https://lists.osgeo.org/pipermail/geonode-devel/2023-August/003335.html) on good practices starting a GeoNode project based  by tracking upstream changes (`geonode-project` detaches your from upstream repo, right?). Bring in the flexibility of using devcontainer and all ways Docker provides was an additional requirement. I tried to encapsulate that huge settings file and tried to set reasonable default values, so the settings look as minimal as possible (there are interdependent variables, so less to fuzz around with).

The blueprint was there before GeoSolutions refactored to use [base images](https://github.com/geonode/geonode-docker). So there was lot of improvement since then. However, coming from Docker context this setup feels a lot more natural to me, than the `geonode-project` setup.

**Reply by t-book** [reply]
thanks a lot for the explanation @ridoo. "tracking upstream changes" this is a  long question on my side as well. Took several tries but haven't found anything that really convinced. I ill give your setup definitly a try. Thanks for sharing!

**Reply by gromit83** [reply]
Could be. I'm not going to argue about any of that. only that I followed this religously. https://docs.geonode.org/en/master/install/advanced/core/index.html
